feat: remove cli/sn_client and faucet API

joshuef opened the pull request:
To reintroduce native tokens now, we should do so encapsulated behind the Wallet struct in the autonomi API. This is used buy ndoes for verification, and clients for paying sending etc. This allows us to keep the payment layer an abstraction in that one place (and optinally enact/use certain types of payment or validation layers.

This commit removes everything in One Big Go, such that we can easily retrieve the state of the native token APIs (which should work) at this point.

(That said, with EVM in, the CI tests were not running here, but it should not be far off working)

joshuef commented on sn_node/src/node/mod.rs:
```diff
@@ -413,10 +414,16 @@ impl MyNode {
             return vec![zero_dkg_id, one_dkg_id];
         }
 
+        let current_elders = BTreeSet::from_iter(sap.elders().copied());
```
This would prevent any splits, right?

maqi replied:
yeah, you are right.
However our current CI tests don't involve any split yet.
will correct it anyway.

oetyng reviewed (commented):
I'm missing a description of the PR.

joshuef committed:
feat: remove cli/sn_client and faucet API

To reintroduce native tokens now, we should do so encapsulated behind the Wallet struct
in the autonomi API. This is used buy ndoes for verification, and clients for paying
sending etc. This allows us to keep the payment layer an abstraction in that one place
(and optinally enact/use certain types of payment or validation layers.

This commit removes everything in One Big Go, such that we can easily retrieve the
state of the native token APIs (which should work) at this point.

(That said, with EVM in, the CI tests were not running here, but it should not
be far off working)
